### PR TITLE
[turn] fix distance measure for equal points

### PR DESCRIPTION
**Pull request by barendgehrels**
Fixes: #1076

### CODE REVIEW

File: test/algorithms/set_operations/difference/difference.cpp [hunk 1]
```diff
@@ -92,15 +92,13 @@ void test_all()
         1, 5, 1.0,
         1, 5, 1.0);
 
-#if defined(BOOST_GEOMETRY_USE_RESCALING) || defined(BOOST_GEOMETRY_TEST_FAILURES)
```
Comment (barendgehrels) [inline test/algorithms/set_operations/difference/difference.cpp:95]: This was already fine now, it is not fixed by this particular fix

File: include/boost/geometry/algorithms/detail/overlay/get_distance_measure.hpp [hunk 2]
```diff
@@ -117,17 +116,32 @@ namespace detail
...
+    // because it is about very tiny differences.
+    auto identical = [](const auto& point1, const auto& point2)
+    {
+        return geometry::get<0>(point1) == geometry::get<0>(point2)
+            && geometry::get<1>(point1) == geometry::get<1>(point2);
+    };
 
+    if (identical(p1, p) || identical(p2, p))
+    {
+        detail::distance_measure<calc_t> const result;
```
Comment (vissarion) [inline include/boost/geometry/algorithms/detail/overlay/get_distance_measure.hpp:136]: is it safer and cleaner to set this explicitly to `0`?

File: test/algorithms/set_operations/difference/difference.cpp (hunk 1 above)
Comment (vissarion) [reply test/algorithms/set_operations/difference/difference.cpp:95]: Sure, but this comment will be lost in the history of commits. Is it better to add this changes in a separate commit? Also applies to other tests in this PR.

**Review by vissarion — APPROVED** [review]
I am OK, thanks!

File: include/boost/geometry/algorithms/detail/overlay/get_distance_measure.hpp (hunk 2 above)
Comment (barendgehrels) [reply include/boost/geometry/algorithms/detail/overlay/get_distance_measure.hpp:136]: It's done in the constructor of `distance_measure`

File: test/algorithms/set_operations/difference/difference.cpp (hunk 1 above)
Comment (barendgehrels) [reply test/algorithms/set_operations/difference/difference.cpp:95]: :+1: I'll do

Comment (barendgehrels) [reply test/algorithms/set_operations/difference/difference.cpp:95]: :heavy_check_mark: done, thanks

File: include/boost/geometry/algorithms/detail/overlay/get_distance_measure.hpp (hunk 2 above)
Comment (vissarion) [reply include/boost/geometry/algorithms/detail/overlay/get_distance_measure.hpp:136]: I know, I just comment that it is more clear to do explicit initialization so the reader will not have to see the declaration of the constructor, but that is my own preference. So I am resolving the comment :)

Comment (barendgehrels) [reply include/boost/geometry/algorithms/detail/overlay/get_distance_measure.hpp:136]: Thanks. Agree it's more clear, but it's internal code and takes (a minimum amount) of time...

**Comment by vissarion** [comment]
@barendgehrels do you want this to be in `1.81`?

**Comment by barendgehrels** [comment]
> > @barendgehrels do you want this to be in `1.81`?
> 
> That would be nice. Is it possible? I can merge it now.

Then I'll remove the doc-part about geographic buffer from this PR.

**Comment by barendgehrels** [comment]
> > > @barendgehrels do you want this to be in `1.81`?
> > 
> > 
> > That would be nice. Is it possible? I can merge it now.
> 
> Then I'll remove the doc-part about geographic buffer from this PR.

:heavy_check_mark: done

**Comment by barendgehrels** [comment]
> > > > @barendgehrels do you want this to be in `1.81`?
> > > 
> > > 
> > > That would be nice. Is it possible? I can merge it now.
> > 
> > 
> > Then I'll remove the doc-part about geographic buffer from this PR.
> 
> heavy_check_mark done

I'll merge it as soon as the checks are green. @awulkiew are you fine with this?

File: doc/release_notes.qbk [hunk 3]
```diff
@@ -23,9 +23,9 @@
 [heading Boost 1.81]
 [/=================]
 
-[*Major improvements]
+[*Solved issues]
 
-* [@https://github.com/boostorg/geometry/pull/1045 1045] Support geographic buffer for linestrings (side, joins)
+* [@https://github.com/boostorg/geometry/issues/1076 1076] Union: in rare cases it might miss one polygon
```
Comment (barendgehrels) [inline doc/release_notes.qbk:28]: @vissarion if you cherry-pick this commit (together with the other 2), then it should be fine.

**Pull request merged**